src/core/server/config/ensure_valid_configuration.ts

```diff
@@ -6,20 +6,13 @@
  * Side Public License, v 1.
  */
 
-import { getUnusedConfigKeys } from './get_unused_config_keys';
-import { ConfigService } from '../../config';
-import { CriticalError } from '../../errors';
-import { LegacyServiceSetupConfig } from '../types';
+import { ConfigService } from '@kbn/config';
+import { CriticalError } from '../errors';
 
-export async function ensureValidConfiguration(
-  configService: ConfigService,
-  { legacyConfig, settings }: LegacyServiceSetupConfig
-) {
-  const unusedConfigKeys = await getUnusedConfigKeys({
-    coreHandledConfigPaths: await configService.getUsedPaths(),
-    settings,
-    legacyConfig,
-  });
+export async function ensureValidConfiguration(configService: ConfigService) {
```

Comment:
Should we convert it into a method of ConfigService? It's odd that `server` imports it from `/config` and passes ConfigService as an argument.